src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -543,14 +545,30 @@ public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseReque
             }
             Assert.Equal(FrameType.Headers, frame.Type);
             HeadersFrame headersFrame = (HeadersFrame) frame;
+            bool endOfStream = (frame.Flags & FrameFlags.EndStream) == FrameFlags.EndStream;
 
-            // TODO CONTINUATION support
-            Assert.Equal(FrameFlags.EndHeaders, FrameFlags.EndHeaders & headersFrame.Flags);
+            using MemoryStream buffer = new MemoryStream();
+            await buffer.WriteAsync(headersFrame.Data);
+
+            // Receive CONTINUATION frames for request.
+            while ((FrameFlags.EndHeaders & frame.Flags) != FrameFlags.EndHeaders)
+            {
+                frame = await ReadFrameAsync(_timeout).ConfigureAwait(false);
+                if (frame == null)
+                {
+                    throw new IOException("Failed to read Continuation frame.");
+                }
+                Assert.Equal(FrameType.Continuation, frame.Type);
```

Comment:
It might be worth it to assert that we aren't setting any other flags here aside from EndHeaders.

Reply:
Added.